### PR TITLE
Added llvm opt pipeline timing info

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
Supporting efforts related to #4283

image

### CODE REVIEW

File: lib/base-compiler.ts [hunk 1]
```diff
@@ -1049,11 +1049,13 @@ export class BaseCompiler {
         const execOptions = this.getDefaultExecOptions();
         execOptions.maxOutput = 1024 * 1024 * 1024;
 
+        const compile_start = performance.now();
```
Comment (partouf) [inline lib/base-compiler.ts:1052]: Can you use the numbers that exec() returns here? To make it more consistent with other timings and not include too much js overhead

File: lib/base-compiler.ts [hunk 2]
```diff
@@ -1049,12 +1049,15 @@ export class BaseCompiler {
...
         execOptions.maxOutput = 1024 * 1024 * 1024;
 
+        const compileStart = performance.now();
         const output = await this.runCompiler(this.compiler.exe, newOptions, this.filename(inputFilename), execOptions);
+        const compileEnd = performance.now();
 
         if (output.timedOut) {
             return {
-                error: 'Compilation timed out',
+                error: 'Clang invocation timed out',
```
Comment (jeremy-rifkin) [inline lib/base-compiler.ts:1058]: Updated wording since it came up on discord

File: types/compilation/compilation.interfaces.ts [hunk 3]
```diff
@@ -99,6 +102,13 @@ export type CompilationResult = {
     jsnesrom?: string;
 
     hints?: string[];
+
+    retreivedFromCache?: boolean;
```
Comment (jeremy-rifkin) [inline types/compilation/compilation.interfaces.ts:106]: I am not 100% confident in these types but they can work for now

File: static/widgets/timing-info-widget.ts [hunk 4]
```diff
@@ -26,14 +26,16 @@ import $ from 'jquery';
...
+import _ from 'underscore';
 
 type Data = ChartData<'bar', number[], string> & {steps: number};
 
 function pushTimingInfo(data: Data, step: string, time: number | string) {
     if (typeof time === 'string') {
         time = parseInt(time, 10);
     }
-    data.labels?.push(`${step} (${time}ms)`);
+    data.labels?.push(`${step} (${Math.round(time * 100) / 100}ms)`);
```
Comment (jeremy-rifkin) [inline static/widgets/timing-info-widget.ts:38]: May as well update this while I'm here

**Comment by jeremy-rifkin** [comment]
Added a timer for `processExecutionResult` even though it's unlikely to ever take much time

image

**Comment by partouf** [comment]
now live

**Pull request merged**